### PR TITLE
[User Management] Fix user management bugs and add loading component : Fix #407

### LINKED ISSUE #407: [User Management] after a user has been added, the row of that user is not yet active and the user cannot be edited
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of the COMET-WEB
- [x] I have searched [open](https://github.com/RHEAGROUP/COMET-WEB-Community-Edition/issues) and [closed](https://github.com/RHEAGROUP/COMET-WEB-Community-Edition/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
after a user has been added, the row of that user is not yet active and the user cannot be edited. The row should immediately be editable

### Steps to Reproduce
  - create a user
  - try to make the user active or inactive by selecting the checkbox -> note that the checkbox and deprecate button are not active

### Logs

  - [x] There are no logs regarding the issue
  
### System Configuration

- COMET version:
  - [x] COMET-WEB: 1.2.0.0

### PR DESCRIPTION
**Pull request by nabilatrhea**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/COMET-WEB-Community-Edition/pulls) open
- [x] I have verified that I am following the COMET-WEB [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/COMET-WEB-Community-Edition/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description

Fix #407 

Fix user management bugs, use the SingleIterationApplicationBase component and add loading component

### CODE REVIEW

File: COMETwebapp/Components/UserManagement/UserManagementTable.razor.cs [hunk 1]
```diff
@@ -174,5 +168,10 @@ protected override async Task OnAfterRenderAsync(bool firstRender)
                 .Subscribe(_ => this.HideOrShowDeprecatedItems()));
             }
         }
+
+        protected override void InitializeValues(Dictionary<string, string> parameters)
+        {
```
Comment (samatrhea) [inline COMETwebapp/Components/UserManagement/UserManagementTable.razor.cs:173]: why do we have an empty method here?

**Review by samatrhea — CHANGES_REQUESTED** [review]

File: COMETwebapp/Components/UserManagement/UserManagementTable.razor.cs (hunk 1 above)
Comment (antoineatrhea) [reply COMETwebapp/Components/UserManagement/UserManagementTable.razor.cs:173]: A super class have this method as abstract, it should be used to initialize Parameters value that come from the URL (like selected option if any).

**Review by antoineatrhea — APPROVED** [review]

File: COMETwebapp/Components/UserManagement/UserManagementTable.razor.cs (hunk 1 above)
Comment (nabilatrhea) [reply COMETwebapp/Components/UserManagement/UserManagementTable.razor.cs:173]: In the UserManagement Page we don't have selectors that needs parameters from the url

Comment (samatrhea) [reply COMETwebapp/Components/UserManagement/UserManagementTable.razor.cs:173]: understood

**Review by samatrhea — APPROVED** [review]

File: COMETwebapp/Components/UserManagement/UserManagementTable.razor.cs (hunk 1 above)
Comment (antoineatrhea) [reply COMETwebapp/Components/UserManagement/UserManagementTable.razor.cs:173]: @nabilatrhea can you add documentation to the method and write a small comment, inside the empty method, why it is empty please?

**Pull request merged**